### PR TITLE
fix: use next pagination across project

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

**Comment by marc0olo** [comment]
thanks @thepiwo!

@AtanasKrondev can you review this PR? it is supposed to address the upcoming mdw backend which dropped support for the page param.

I assume you can check by using `https://testnet.aeternity.art/mdw` as backend. the new version isn't deployed officially yet

**Comment by AtanasKrondev** [comment]
``Load more`` on transactions page wasn't working when the type is ``All``.  And the use of objects on the same page to check if a transaction is already fetched seems redundant. @thepiwo @marc0olo  check my commits about this.
Also seems like that the transactions module in the store needs refactoring, as we are only calling actions without relying on the state, but that may be a different topic.

**Comment by marc0olo** [comment]
the only way I can test it is clicking through the app and check about things not working. I will do that asap. will compare with current state of mdw which is deployed.

> Also seems like that the transactions module in the store needs refactoring, as we are only calling actions without relying on the state, but that may be a different topic.

is this about remembering the cursor which would allow to return to the correct position in overview when visiting a tx detailview?

**Comment by AtanasKrondev** [comment]
I mean the vuex store. We are not using any state, we only use the actions to make the calls to the mdw. It's just an observation, I need to make a research if it'll be better to save the already fetched transactions in it... although due to the dynamic nature of the constantly updating transaction list probably it's not needed. Or would it be better to organise in a different way the data fetching.

File: store/transactions.js [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
-      }
-    }
+    const transformed = transactions.map(t => t.tx.type === 'GAMetaTx' ? transformMetaTx(t) : t)
+    state.transactions = [...transactions, transformed]
   }
 }
 
 export const actions = {
-  getLatest: async function ({ state, rootGetters: { middleware }, commit }, { limit }) {
+  getLatest: async function ({ state, rootGetters: { middleware }, commit }, { limit, page }) {
```
Comment (thepiwo) [inline store/transactions.js:15]: I don't fine where getLatest is called with page

**Comment by thepiwo** [comment]
I tried to not include any other refactorings in here other than the hacky switch to use next page instead of page + 1. All this needs to be refactored to be more like some of the previously existing nextPage logic in store.

**Review by AtanasKrondev — APPROVED** [review]
For me the PR is good to go

**Comment by marc0olo** [comment]
@thepiwo @AtanasKrondev this is intended to be used with `v1.3.0` of the middleware, right?

I am currently testing with following URLS:
```
    middlewareURL: process.env.NUXT_APP_NODE_URL || 'https://mainnet.test.aeternity.art/mdw',
    middlewareWS: process.env.NUXT_APP_NODE_WS || 'wss://mainnet.test.aeternity.art/mdw/websocket',
```

using these URLs I get an error on the dashboard, here the console log:
image

**Comment by AtanasKrondev** [comment]
Seems to work fine with the current mainnet version (no changes at all to the env)

**Comment by marc0olo** [comment]
@AtanasKrondev today I tested again with the current mainnet version. I get the same error on the dashboard. can you please check and click around, go back to dashboard, etc.?

I can also provide a screencast. I see the latest generations loading and suddenly an error happens.

I am talking about dashboard view only

**Review by marc0olo — CHANGES_REQUESTED** [review]
explicitely requesting the changes just that we don't forget to fix. from my latest information Atanas also faced the issue on dashboard

**Review by marc0olo — APPROVED** [review]
looks good to me know. also works with `v1.3.1` of mdw backend ;-)

**Pull request merged**